pkgs/development/tools/click/default.nix

```diff
@@ -0,0 +1,92 @@
+{ lib
+, fetchFromGitLab
+, buildPythonApplication
+, autoreconfHook
+, debian
+, automake
+, perl
+, libtool
+, vala
+, pkg-config
+, libgee
+, json-glib
+, properties-cpp
+, gobject-introspection
+, getopt
+, setuptools
+, pygobject3
+, wrapGAppsHook
+}:
+
+buildPythonApplication {
+  pname = "click";
+  version = "22-02-2023";
+
+  src = fetchFromGitLab {
+    owner = "ubports";
+    repo = "development/core/click";
+    rev = "aaf2735e8e6cbeaf2e429c70136733513a81718a";
+    sha256 = "sha256-pNu995/w3tbz15QQVdVYBnWnAoZmqWj1DN/5PZZ0iZw=";
+  };
+
+  configureFlags = [
+    "--with-systemdsystemunitdir=${placeholder "out"}/lib/systemd/system"
+    "--with-systemduserunitdir=${placeholder "out"}/usr/lib/systemd/user"
+  ];
+
+  preConfigure = ''
+    export click_cv_perl_vendorlib=$out/${perl.libPrefix}
+    export PYTHON_INSTALL_FLAGS="--prefix=$out"
+  '';
+
+  installPhase = ''
+    mkdir -p $out
+    cp -r {bin,click_package,build,lib} $out
+    # cp -r . $out
+  '';
```

Comment:
Don't do this. You're skipping the software's own install methods, which should work perfectly fine with the set flags & variables.

```
installing
install flags: SHELL=/nix/store/561wgc73s0x1250hrgp7jm22hhv7yfln-bash-5.2-p15/bin/bash gsettingsschemadir=/nix/store/ajkwg8bra667by9b4ag2ninx79pvdjbx-click-unstable-2023-02-22/share/gsettings-schemas/click-unstable-2023-02-22/glib-2.0/schemas/ gsettingsschemadir=/nix/store/ajkwg8bra667by9b4ag2ninx79pvdjbx-click-unstable-2023-02-22/share/gsettings-schemas/click-unstable-2023-02-22/glib-2.0/schemas/ gsettingsschemadir=/nix/store/ajkwg8bra667by9b4ag2ninx79pvdjbx-click-unstable-2023-02-22/share/gsettings-schemas/click-unstable-2023-02-22/glib-2.0/schemas/ install
Making install in lib
make[1]: Entering directory '/build/source/lib'
Making install in click
make[2]: Entering directory '/build/source/lib/click'
make  install-am
make[3]: Entering directory '/build/source/lib/click'
make[4]: Entering directory '/build/source/lib/click'
 /nix/store/7w1s50i8yplw2gkpf9hpha6n9vci8g98-coreutils-9.1/bin/mkdir -p '/nix/store/ajkwg8bra667by9b4ag2ninx79pvdjbx-click-unstable-2023-02-22/lib'
 /nix/store/561wgc73s0x1250hrgp7jm22hhv7yfln-bash-5.2-p15/bin/bash ../../libtool   --mode=install /nix/store/7w1s50i8yplw2gkpf9hpha6n9vci8g98-coreutils-9.1/bin/install -c   libclick-0.4.la '/nix/store/ajkwg8bra667by9b4ag2ninx79pvdjbx-click-unstable-2023-02-22/lib'
libtool: install: /nix/store/7w1s50i8yplw2gkpf9hpha6n9vci8g98-coreutils-9.1/bin/install -c .libs/libclick-0.4.so.0.4.0 /nix/store/ajkwg8bra667by9b4ag2ninx79pvdjbx-click-unstable-2023-02-22/lib/libclick-0.4.so.0.4.0
```

```suggestion
```